README.md

```diff
@@ -15,30 +15,109 @@
+If you're familiar with Home Manager, think of it as similar to Home Manager but for your entire machine. Whereas Home Manager manages user environments, System Manager manages the full system, starting at root-level configurations, packages, and services, using the same reliable, Nix-based model.
 
-## Usage
+System Manager builds on the many modules that already exist in [NixOS].
 
-### Install Nix
+## Quick Example to Get Started
 
-In order to use System Manager, you will first need to install Nix.
-You can either use your distro's package manager, or use one of the available options
-to install Nix.
+Assume you're using a non-NixOS distrubution (such as Ubuntu) and you have Nix already installed.
 
-- [Official Nix Installer][official-installer] - The canonical source for installing nix.
-- [Determinate Nix Installer][detsys-installer] - A wrapper around the official installer that has SELinux support, and enables flake features by default.
+In a folder create a file called `flake.nix` with the following:
+
+```nix
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+
+    system-manager = {
+      url = "github:numtide/system-manager";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, system-manager }: {
+    systemConfigs.default = system-manager.lib.makeSystemConfig {
+      modules = [
+        ./modules
+      ];
+    };
+  };
+}
+```
+
+In that folder create a subfolder called `modules`. In `modules` create a file called
+`default.nix` with the following system configuration:
+
+```nix
+{ pkgs, ... }:
+{
+  nixpkgs.hostPlatform = "x86_64-linux";
+  
+  environment.systemPackages = with pkgs; [
+    btop          # Beautiful system monitor
+    bat           # Modern 'cat' with syntax highlighting
+  ];
+}
 
-## Usage with flakes
+```
 
-### Defining the configuration
+This specifies a configuration that includes btop and bat as installed on the system. (In other words, system manager will install them on your system.) To do so, execute system manager with sudo using the nix command (assuming you have experimental features nix-command and flakes turned on):
 
-A basic Nix flake using System Manager would look something like this:
+```
+sudo env PATH="$PATH" nix run 'github:numtide/system-manager' -- switch --flake .
+```
+
+Notice we're passing the current PATH environment into sudo so that the elevated shell can locate the nix command.
+
+Also, note that you might need to enable nix-commands and flakes:
+
+```
+sudo env PATH="$PATH" nix --extra-experimental-features 'nix-command flakes' run 'github:numtide/system-manager' -- switch --flake .
+```
+
+> [!Note]
+> The first time you run system manager, it will update your path by adding an entry inthe /etc/profile.d folder. For such change to take effect, you need to log out and log back in. However, if you don't want to log out, you can simply source the file:
```

Comment:
```suggestion
> The first time you run system manager, it will update your path by adding an entry in the /etc/profile.d folder. For this change to take effect, you need to log out and then log back in. However, if you don't want to log out, you can source the file:
```